### DISCUSSION TITLE
Does Yabai support a grid style virtual desktop layout similar to TotalSpace2 on Mac, or FVWM on linux/unix?

### DISCUSSION THREAD

**Discussion by signal15** [asmvik/yabai#1683] [Q&A] [answered]
TotalSpaces2 doesn't work on Apple silicon, and TotalSpaces3 seems to be abandoned.  The alphas that have leaked, don't really work.  If anyone remembers back that far, the original virtual desktop implementation in the early 2000's on Mac was a grid until they changed it into this stupid linear row of all your desktops.

I want to be able to specify a 3x3 (or whatever) grid.  And then switch between with Command+arrow keys.  For example, in a 3x3 grid, if I was on Desktop 4 (middle row left), and pressed command+up arrow, it would go to Desktop 1 (top row, left column).

Is this possible?

Note that I'm not looking for a grid on one screen that snaps my apps into a specific size.  I'm looking for virtual desktops/screens that are organized in a grid.

**Comment by asmvik** [comment]
All yabai does is make the space transition instant, as well as exposing functionality for manipulating spaces through a command line interface. The grid system can probably be scripted on top of this interface.

**Comment by 24unix** [comment]
> TotalSpaces2 doesn't work on Apple silicon, and TotalSpaces3 seems to be abandoned. 

Not yet. I've got it running here.

<img width="498" alt="image" src="https://user-images.githubusercontent.com/36622631/228550007-212a5043-0058-441d-a3a3-c202751d9218.png">


However, unlike in version 2 "Sync Grid" is not available, so it's currently useless.
I asked the support at the weekend if we can expect the feature to come back, but got no reply yet.

<img width="541" alt="image" src="https://user-images.githubusercontent.com/36622631/228550679-0add5f80-3c4d-4ff9-87d6-1bcb062f42c9.png">

**Comment by signal15** [comment]
I have the same version.  It doesn't seem to work reliably and stops
switching after my laptop sleeps and wakes back up, or things get wonky
when I plug in a second monitor.  Things that TF2 handled perfectly.  I
just need something reliable, and if yabai will do it with some scripting,
then I guess I'll have to do some scripting to make it happen.

On Wed, Mar 29, 2023 at 8:20 AM Micha ***@***.***> wrote:

> TotalSpaces2 doesn't work on Apple silicon, and TotalSpaces3 seems to be
> abandoned.
>
> Not yet. I've got it running here.
>
> [image: image]
> <https://user-images.githubusercontent.com/36622631/228550007-212a5043-0058-441d-a3a3-c202751d9218.png>
>
> However, unlike in version 2 "Sync Grid" is not available, so it's
> currently useless.
> I asked the support at the weekend if we can expect the feature to come
> back, but got no reply yet.
>
> [image: image]
> <https://user-images.githubusercontent.com/36622631/228550679-0add5f80-3c4d-4ff9-87d6-1bcb062f42c9.png>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/koekeishiya/yabai/discussions/1683#discussioncomment-5465832>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADEUHG32HTH6GRGQJYWUZILW6QZKZANCNFSM6AAAAAAWLBWV6A>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
--
jay austad  |  612.423.1433  |  ***@***.***

**Comment by 24unix** [comment]
I don't use notebooks with external displays, so that it no issue for me, my iMac always has two external displays attached.

I hope BinaryAge will get things fixed. Else I might be into completing my own yabai frontend, but as i currently switched the job time is limited. That was the last update, then I ran put of time: https://github.com/koekeishiya/yabai/discussions/759

But: TS3 can perfectly switch spaces, but limited to the current display. So technically it should be possible to reenable the sync thing.

**Comment by asmvik** [comment, marked as answer]
I found this with a quick search, but haven't tried it myself: https://github.com/mikkelricky/grid-spaces
It is apparently an attempt at a grid system built on top of the spaces interface exposed by yabai.

**Reply by mikkelricky** [reply]
A friend of mine just referred me to this thread. I've implemented “2D grid navigation” in yabai and have been using it for quite some time. Now I've made a feature request (and an accompanying pull request) on adding it to yabai: https://github.com/koekeishiya/yabai/issues/2602
